azuread/resource_application.go

```diff
@@ -297,6 +350,23 @@ func resourceApplicationUpdate(d *schema.ResourceData, meta interface{}) error {
 		properties.RequiredResourceAccess = expandADApplicationRequiredResourceAccess(d)
 	}
 
+	if d.HasChange("app_role") {
+		// if the app role already exists then it must be disabled
+		// with no other changes before it can be edited or deleted
+		var appRolesProperties graphrbac.ApplicationUpdateParameters
+		oldAppRolesRaw, newAppRolesRaw := d.GetChange("app_role")
+		appRolesProperties.AppRoles = expandADApplicationAppRoles(oldAppRolesRaw.(*schema.Set).List())
```

Comment:
This doesn't account for if the app role has changed on the azure side. Would it make more sense to do a client.Get() to get the real state of the world here so the disablement won't fail?

Reply:
I'll take a look into this

Reply:
think that should be sorted now